examples/sp/sp_tutorial.py

```diff
@@ -0,0 +1,312 @@
+print "---------------------------------"
+print ""
+
+plt.plot(x, y)
+plt.xlabel("Noise level")
+plt.ylabel("Input overlap")
+plt.title("Figure 2")
+plt.show()
+
+# Part 2b: 
+# -------
+# The output overlap between two binary input vectors is the overlap of the
+# columns that become active once they are fed to the SP. In this part we
+# turn learning off, and observe the output of the SP as we input two binary 
+# input vectors with varying level of noise.
+# Starting from two identical vectors (that yield the same active columns)
+# we would expect that as we add noise to one of them their output overlap
+# decreases.
+# In this part we will show how the output overlap behaves in function of the
+# input overlap between two vectors.
+
+x = []
+y = []   
+for noiseLevel in np.arange(0, 1.1, 0.1):
+  resetVector(inputX1, inputX2)
+  corruptVector(inputX2, noiseLevel)
+  
+  sp.compute(inputX1, False, outputX1)
+  sp.compute(inputX2, False, outputX2)  
+  
+  x.append(pctOverlap(inputX1, inputX2, inputSize))
+  y.append(pctOverlap(outputX1, outputX2, columnNumber))
+
+print ""
+print "---------------------------------"
+print "Figure 3 shows the output overlap between two sparse representations" 
+print "in function of their input overlap. Starting from two identical binary vectors"
+print "(which yield the same active columns) we add noise two one of them"
+print "feed it to the SP, and estimate the output overlap between the two"
+print "representations in terms of the common active columns between them."
+print "As expected, as the input overlap decrease, so does the output overlap."
+print "However, this relationship is sublinear, implying that the SP is very"
+print "sensitive to noise."
+print "---------------------------------"
+print ""
+
+plt.plot(x, y)
+plt.xlabel("Input overlap")
+plt.ylabel("Output overlap")
+plt.title("Figure 3")
+plt.show()
+
+# Part 3: 
+# -------
+# As we saw in the last section, an untrained SP is very sensitive to noise in a binary vector
+# in such a way that the output overlap between a vector and a corrupted version of it decreases
+# dramatically as the amount of noise grows. To improve this situation we train the SP by
+# turning learning on, and by exposing it to a variety of random binary vectors.
+# We will expose the SP to a repetition of input patterns in order to make it learn and distinguish
+# them once learning is over. This will result in robustness to noise in the inputs.
+# In this section we will reproduce the plot in the last section after the SP has learned a series
+# of inputs. Here we will see how the SP exhibits some resilience to noise after learning.
+
+# We will present 10 random vectors to the SP, and repeat this 30 times.
+# Try changing the number of times we do this to see how it changes the last plot.
+# Then, you could also modify the number of examples to see how the SP behaves.
+# Is there a relationship between the number of examples and the number of times that
```

Comment:
yes, I was thinking that the user could play around with those parameters in order to gain more insight about the SP. What do you think?